scipy/linalg/_basic.py

```diff
@@ -163,7 +163,8 @@ def solve(a, b, lower=False, overwrite_a=False,
 
     # accommodate empty arrays
     if b1.size == 0:
-        return np.asfortranarray(b1.copy())
+        dt = solve(np.eye(2, dtype=a1.dtype), np.ones(2, dtype=b1.dtype)).dtype
```

Comment:
Solving with small non-empty arrays is a handy trick; should be more robust that getting this right by calling something like `np.result_type`.